train_model.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/python
+import time
+import argparse
+from sklearn import linear_model
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.neighbors import KNeighborsClassifier
+import numpy as np
+from termcolor import colored
+import pickle
+
+def parse(line):
+    tokens = line.split(',')
+    features = [tokens[3], tokens[4], tokens[5], tokens[13], tokens[14], tokens[15], tokens[16], 
+        tokens[17], tokens[20], tokens[21], tokens[22]]
+    features = [float (f) for f in features]
+    label = int (tokens[23])
```

Comment:
Please add an optional argument specifying whether you are parsing labeled (train) data or you are parsing unlabeled (test) data.

`def parse(line, is_labeled=True):`

if this argument is `False` then you'll need to `return (features, None)` and not try to parse the label.

Please also propagate it `load_dataset_from_file` function discussed in comment below so you have `def load_dataset_from_file(filename, examples_count, is_labeled=True):`. It should work well with both labeled and unlabeled data and reused between train_model.py and predict_model.py.